MSAL/test/app/mac/MSALAcquireTokenViewController.m

```diff
@@ -0,0 +1,366 @@
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+#import "MSALAcquireTokenViewController.h"
+#import <MSAL/MSAL.h>
+#import "MSALTestAppSettings.h"
+#import "MSALScopesViewController.h"
+#import "MSALInteractiveTokenParameters.h"
+#import "MSALPublicClientApplication+Internal.h"
+#import "MSIDDefaultTokenCacheAccessor.h"
+#import "MSALSilentTokenParameters.h"
+#import "WebKit/WebKit.h"
+
+static NSString * const clientId = @"clientId";
+static NSString * const redirectUri = @"redirectUri";
+static NSString * const defaultScope = @"User.Read";
+
+@interface MSALAcquireTokenViewController ()
+
+@property (weak) IBOutlet NSPopUpButton *profiles;
+@property (weak) IBOutlet NSTextField *clientId;
+@property (weak) IBOutlet NSTextField *redirectUri;
+@property (weak) IBOutlet NSTextField *scopesLabel;
+@property (weak) IBOutlet NSSegmentedControl *promptBehavior;
+@property (weak) IBOutlet NSTextField *loginHintField;
+@property (weak) IBOutlet NSTextView *resultView;
+@property (weak) IBOutlet NSTextField *extraQueryParamsField;
+@property (weak) IBOutlet NSSegmentedControl *webViewType;
+@property (weak) IBOutlet NSSegmentedControl *validateAuthority;
+@property (weak) IBOutlet NSStackView *acquireTokenView;
+@property (weak) IBOutlet WKWebView *webView;
+
+
+@property MSALTestAppSettings *settings;
+@property NSArray *selectedScopes;
+
+@end
+
+@implementation MSALAcquireTokenViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    self.settings = [MSALTestAppSettings settings];
+    [self populateProfiles];
+    self.selectedScopes = @[defaultScope];
+}
+
+- (void)populateProfiles
+{
+    [self.profiles removeAllItems];
+    [self.profiles setTarget:self];
+    [self.profiles setAction:@selector(selectedProfileChanged:)];
```

Comment:
If this button is added in storyboard, the action can be hooked up there as well

Reply:
Can we add this action in storyboard instead?

Reply:
Done